### PR TITLE
Sync waiters with latest versions

### PR DESCRIPTION
**Pull request by jamesls**

### CODE REVIEW

**Comment by dstufft** [comment]
Did you mean to set ``botocore/data/dynamodb/2012-08-10/waiters-2.json`` to executable?

**Comment by jamesls** [comment]
Pushed a commit that removed the executable bit from all our json files.

File: botocore/data/appstream/2016-12-01/waiters-2.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{
+  "version": 2,
+  "waiters": {
+    "FleetStarted": {
```
Comment (kyleknap) [inline botocore/data/appstream/2016-12-01/waiters-2.json]: Are these correct? Looking at the [docs](https://docs.aws.amazon.com/appstream2/latest/APIReference/API_Fleet.html#AppStream2-Type-Fleet-State). It looks like the states should be `RUNNING` and `STOPPED`. There are also no `PENDING_*` states.

File: botocore/data/elbv2/2015-12-01/waiters-2.json [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      "maxAttempts": 40,
+      "acceptors": [
+        {
+          "state": "success",
+          "matcher": "pathAll",
+          "argument": "LoadBalancers[].State.Code",
+          "expected": "active"
+        },
+        {
+          "state": "retry",
```
Comment (kyleknap) [inline botocore/data/elbv2/2015-12-01/waiters-2.json]: This state is unneeded, but should not block pulling it in.

File: botocore/data/elbv2/2015-12-01/waiters-2.json [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        }
+      ]
+    },
+    "LoadBalancersDeleted": {
+      "delay": 15,
+      "operation": "DescribeLoadBalancers",
+      "maxAttempts": 40,
+      "acceptors": [
+        {
+          "state": "retry",
```
Comment (kyleknap) [inline botocore/data/elbv2/2015-12-01/waiters-2.json]: Another unneeded acceptor. Would be nice to fix but should not block pulling this in.

File: botocore/data/rds/2014-10-31/waiters-2.json [hunk 4]
```diff
@@ -98,20 +86,87 @@
         }
       ]
     },
-    "DBSnapshotCompleted": {
```
Comment (kyleknap) [inline botocore/data/rds/2014-10-31/waiters-2.json]: This looks like a breaking change as we are renaming the waiter from `DBSnapshotCompleted` to `DBSnapshotAvaliable`. However, I am not sure how useful it was given its configuration was this:
```
    "DBSnapshotCompleted": {
      "delay": 15,
      "operation": "DescribeDBSnapshots",
      "maxAttempts": 40,
      "acceptors": [
        {
          "expected": "DBSnapshotNotFound",
          "matcher": "error",
          "state": "success"
        },
        {
          "expected": "available",
          "matcher": "pathAll",
          "state": "success",
          "argument": "DBSnapshots[].Status"
        }
      ]
}
```
Looks like the first acceptor should have been a failure?

File: botocore/data/appstream/2016-12-01/waiters-2.json (hunk 1 above)
Comment (jamesls) [reply botocore/data/appstream/2016-12-01/waiters-2.json]: Yeah it's correct.

File: botocore/data/rds/2014-10-31/waiters-2.json (hunk 4 above)
Comment (jamesls) [reply botocore/data/rds/2014-10-31/waiters-2.json]: Ah, I missed that one.  This is one botocore had that wasn't upstream.  I'm going to just add it back verbatim.

**Comment by jamesls** [comment]
@kyleknap Added back the missing waiter.  Just to clarify the intent of this PR, this is syncing all the waiters we can while avoiding ones that are flat out wrong.  Small cleanups aren't something I'm targeting here.  Those ideally happen upstream.

**Pull request merged**